tests/lib/thrift/ThriftTest/src/client.cpp

```diff
@@ -0,0 +1,509 @@
+  response = testClient.testEnum(Numberz::TWO);
+  zassert_equal(response, Numberz::TWO, NULL);
+
+  response = testClient.testEnum(Numberz::EIGHT);
+  zassert_equal(response, Numberz::EIGHT, NULL);
+}
+
+static void typedef_test(void) {
+  ThriftTestClient& testClient = *(context.client);
+  UserId uid = testClient.testTypedef(309858235082523LL);
+  zassert_equal(uid, 309858235082523LL, "Unexpected uid: %llu", uid);
+}
+
+static void nested_map_test(void) {
+  ThriftTestClient& testClient = *(context.client);
+  map<int32_t, map<int32_t, int32_t>> mm;
+  testClient.testMapMap(mm, 1);
+
+  zassert_equal(mm.size(), 2, NULL);
+  zassert_equal(mm[-4][-4], -4, NULL);
+  zassert_equal(mm[-4][-3], -3, NULL);
+  zassert_equal(mm[-4][-2], -2, NULL);
+  zassert_equal(mm[-4][-1], -1, NULL);
+  zassert_equal(mm[4][4], 4, NULL);
+  zassert_equal(mm[4][3], 3, NULL);
+  zassert_equal(mm[4][2], 2, NULL);
+  zassert_equal(mm[4][1], 1, NULL);
+}
+
+static void insanity_test(void) {
+  ThriftTestClient& testClient = *(context.client);
+  Insanity insane;
+  insane.userMap.insert(make_pair(Numberz::FIVE, 5));
+  insane.userMap.insert(make_pair(Numberz::EIGHT, 8));
+  Xtruct truck;
+  truck.string_thing = "Goodbye4";
+  truck.byte_thing = 4;
+  truck.i32_thing = 4;
+  truck.i64_thing = 4;
+  Xtruct truck2;
+  truck2.string_thing = "Hello2";
+  truck2.byte_thing = 2;
+  truck2.i32_thing = 2;
+  truck2.i64_thing = 2;
+  insane.xtructs.push_back(truck);
+  insane.xtructs.push_back(truck2);
+
+  map<UserId, map<Numberz::type, Insanity>> whoa;
+  testClient.testInsanity(whoa, insane);
+
+  zassert_equal(whoa.size(), 2, NULL);
+
+  map<UserId, map<Numberz::type, Insanity>>::const_iterator it1 = whoa.find(UserId(1));
+  zassert_not_equal(it1, whoa.end(), NULL);
+
+  if (it1 != whoa.end()) {
+    auto it12 = it1->second.find(Numberz::TWO);
+    zassert_not_equal(it12, it1->second.end(), NULL);
+    zassert_equal(it12->second, insane, NULL);
+
+    auto it13 = it1->second.find(Numberz::THREE);
+    zassert_not_equal(it13, it1->second.end(), NULL);
+    zassert_equal(it13->second, insane, NULL);
+  }
+
+  map<UserId, map<Numberz::type, Insanity>>::const_iterator it2 = whoa.find(UserId(2));
+  zassert_not_equal(it2, whoa.end(), NULL);
+  if (it2 != whoa.end()) {
+    auto it26 = it2->second.find(Numberz::SIX);
+    zassert_not_equal(it26, it2->second.end(), NULL);
+    zassert_equal(it26->second, Insanity(), NULL);
+  }
+}
+
+static void exception_test(void) {
+  ThriftTestClient& testClient = *(context.client);
+  try {
+    testClient.testException("Xception");
+    zassert_unreachable("Xception: Unhandled exception should abort execution");
+  } catch (Xception& e) {
+  }
+
+  try {
+    testClient.testException("TException");
+    zassert_unreachable("TException: Unhandled exception should abort execution");
+  } catch (const TException&) {
+  }
+
+  try {
+    testClient.testException("success");
+  } catch (exception& ex) {
+    zassert_exception("%s", ex.what());
+  }
```

Comment:
I would use this pattern to check if an exception was thrown:
```suggestion
  std::exception_ptr eptr = nullptr;
  try {
    testClient.testException("success");
  } catch (...) {
    eptr = std::current_exception();
  }
  
  zassert_not_equal(nullptr, eptr, "an exception was not thrown");
```

Reply:
`testException("success")` was supposed to not throw an exception. I think you meant to apply this change to two try-catch blocks above?